interface/modules/custom_modules/oe-module-weno/src/Bootstrap.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+namespace OpenEMR\Modules\WenoModule;
+
+use OpenEMR\Modules\WenoModule\WenoGlobalConfig;
+use OpenEMR\Common\Logging\SystemLogger;
+use OpenEMR\Core\Kernel;
+use OpenEMR\Events\Globals\GlobalsInitializedEvent;
+use OpenEMR\Services\Globals\GlobalSetting;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Twig\Environment;
+use OpenEMR\Menu\MenuEvent;
+use OpenEMR\Events\PatientDemographics\RenderEvent as pRenderEvent;
+use OpenEMR\Events\PatientDemographics\RenderPharmacySectionEvent;
+use OpenEMR\Events\Patient\PatientBeforeCreatedAuxEvent;
+use OpenEMR\Events\Patient\PatientUpdatedEventAux;
+use OpenEMR\Modules\WenoModule\Services\SelectedPatientPharmacy;
+
+class Bootstrap {
+
+    const OPENEMR_GLOBALS_LOCATION = "../../../../globals.php";
+    const MODULE_INSTALLATION_PATH = "/interface/modules/custom_modules/oe-module-weno";
+    const MODULE_NAME = "";
+    const MODULE_MENU_NAME = "Weno";
+
+    /**
+     * @var EventDispatcherInterface The object responsible for sending and subscribing to events through the OpenEMR system
+     */
+    private $eventDispatcher;
+
+    private $moduleDirectoryName;
+
+    /**
+     * The OpenEMR Twig Environment
+     * @var Environment
+     */
+    private $twig;
+
+    /**
+     * @var WenoGlobalConfig
+     */
+    private $globalsConfig;
+
+    /**
+     * @var SystemLogger
+     */
+    private $logger;
+
+    private $modulePath;
+
+    /**
+     * @var SelectedPatientPharmacy
+     */
+    private $selectedPatientPharmacy;
+
+    public function subscribeToEvents()
+    {
+        $this->addGlobalSettings();
+        $this->registerMenuItems();
+        $this->registerDemographicsEvents();
+        $this->demographicsSelectorEvents();
+        $this->demographicsDisplaySelectedEvents();
+        $this->patientSaveEvents();
+        $this->patientUpdateEvents();
+    }
+
+    public function __construct(EventDispatcher $dispatcher, ?Kernel $kernel = null)
+    {
+        global $GLOBALS;
+
+        if (empty($kernel)) {
+            $kernel = new Kernel();
+        }
+        $this->eventDispatcher = $dispatcher;
+
+        $this->globalsConfig = new WenoGlobalConfig($GLOBALS);
+        $this->moduleDirectoryName = basename(dirname(__DIR__));
+        $this->modulePath = dirname(__DIR__);
+        $this->logger = new SystemLogger();
+        $this->selectedPatientPharmacy = new SelectedPatientPharmacy();
+    }
+
+    /**
+     * @return \Twig\Environment
+     */
+    public function getTwig()
+    {
+        return $this->twig;
+    }
+
+    public function addGlobalWenoSettings(GlobalsInitializedEvent $event)
+    {
+        global $GLOBALS;
```

Comment:
This kind of bugs me when used in this context. Globals(supers) have scope almost always in classes(I've never seen otherwise in openemr. Take a $_REQUEST super as example). If not, need to look to the class init scope. I don't think memory impact is significant and really no harm otherwise but, seems out of place and iffy.
It won't stop merge though.